src/HERMIT/API/Dictionary/KURE.hs

```diff
@@ -1,31 +1,89 @@
 {-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
 {-# LANGUAGE OverloadedStrings #-}
 {-# LANGUAGE ScopedTypeVariables #-}
-module HERMIT.API.Dictionary.KURE where
+
+module HERMIT.API.Dictionary.KURE (
+      id
+    , Id
+    , success
+    , fail
+    , (<+)
+    , (>>>)
+    , (>+>)
+    , try
+    , repeat
+    , replicate
+    , all
+    , any
+    , one
+    , allBU
+    , allTD
+    , allDU
+    , anyBU
+    , anyTD
+    , anyDU
+    , oneTD
+    , oneBU
+    , pruneTD
+    , innermost
+    , focus
+    , focusWithPath
+    , when
+    , not
+    , invert
+    , forward
+    , backward
+    , test
+    , anyCall
+    , promote
+    , extractR
+    , extractT
+    , between
+    , atPath
+    , AtPathArgs
+    , serialise
+    ) where
 
 import Data.Aeson
 import Data.Proxy
 
 import HERMIT.API.Types
 
--- | Perform an identity rewrite.
-idCore :: Rewrite LCore
-idCore = Transform $ method "idCore" []
-
--- | Perform an identity rewrite.
-idCoreTC :: Rewrite LCoreTC
-idCoreTC = Transform $ method "idCoreTC" []
+import Prelude hiding
+    ( all
+    , any
+    , fail
+    , id
+    , not
+    , repeat
+    , replicate
+    )
+
+-- |
+-- id :: Rewrite LCore
+--   Perform an identity rewrite.
+-- id :: Rewrite LCoreTC
+--   Perform an identity rewrite.
+id :: Id a => Rewrite a
+id = Transform $ method "idCore" []
+
+-- | Class of types that can have an identity rewrite via 'idCore'.
+class Id a
+instance Id LCore
+instance Id LCoreTC
```

Comment:
This seems strange. Can we not just do away with this class (which has no methods) and make `id` fully polymorphic?